test/CustomHashSetWithMemoryLimitTest.cpp

```diff
@@ -0,0 +1,229 @@
+#include <gtest/gtest.h>
+
+#include "../src/util/HashSet.h"
+#include "util/AllocatorWithLimit.h"
+#include "util/MemorySize/MemorySize.h"
+
+using ad_utility::makeAllocationMemoryLeftThreadsafeObject;
+using namespace ad_utility::memory_literals;
+
+using Set = ad_utility::CustomHashSetWithMemoryLimit<int>;
+
```

Comment:
```suggestion
// ____________________________________________________________________________________________
```

(For readability, same for all the tests.)